### ISSUE TITLE
Maintain a WRC section on the website

### ISSUE THREAD

**Issue by Laura-O** [thewca/wca-regulations#419]
This is not new, but I think we should start to plan more details how the WRC can add more content to the WCA website.

My basic ideas:
* Landing page with links to regulations, guidelines, translations and latest WRC announcements
* FAQ section (tbd, either on the same page or a link)
* Link to organizer guidelines (tbd)

Let's collect some ideas and discuss them. And if a WRC member would like to take over this task and coordinate, collect, etc. everything with the other members, please raise your hand. 😄 

Related issues:
#72 Announce WRC changes
#168 Example puzzle page
#393 Releasing WRC news

**Comment by AlbertoPdRF** [comment]
As always, I'm happy to raise my hand! :)

Your basic ideas seem like a good starting point to me, and I'm sure that other ideas will come up while we work on these.

**Comment by viroulep** [comment]
cc @jfly so that he may follow this discussion.

**Comment by AlbertoPdRF** [comment]
Can we have a template of the WRC section on which we can work in order to start creating and structuring content?

Also, I've created the documents "WRC FAQ" and "Organizer guidelines" in our drive, and I'll start working on them when I have the time. Of course, any help is appreciated (especially pointing out FAQs) :)

**Comment by viroulep** [comment]
> Can we have a template of the WRC section on which we can work in order to start creating and structuring content?

I think it's pretty much up to us: we can imagine any kind of design we would like; if we're gonna have a dedicated WRC section on the website the only piece of layout imposed will be the navigation bar and the footer.
ie this:
![empty-template](https://cloud.githubusercontent.com/assets/1007485/24248799/2418020c-0fd1-11e7-890c-7213e754fb66.png)

So if you have some layout ideas you can pretty much draw them and I can try to put actual components to see how it would look like.

**Comment by AlbertoPdRF** [comment]
Here I am with an idea:

![draft1](https://cloud.githubusercontent.com/assets/26309166/24841169/bebc8468-1d7d-11e7-8101-fd31f526810b.jpg)

I think it's pretty simple, and that's why I like it :P I'll explain it a little in case that it's not clear (my drawing skills are not the best)

So, we'll have images of all the cubes ordered for instance by release date and, when an image is selected, it goes bigger and displays the name of the cube, a green or red frame depending on if it's legal or not, and maybe a little explanation below all the images but still inside the "box" where all the images are. The "box" is intended to be the same as we use for the main page of the WCA, just to keep the website homogeneous. Also, the frame and the names of the cubes could be displayed for all the pictures, and not only for the bigger one.

What do you think about it?

**Comment by lgarron** [comment]
> What do you think about it?

I think it won't be easy to navigate, especially on a phone.

We should also not focus on specific cube types, but archetypes of edge cases.

I suggest copying the Do/Don't layout from Material Design: https://material.io/guidelines/motion/movement.html#movement-movement-within-screen-bounds

**Comment by Laura-O** [comment]
Thanks for making this draft, @AlbertoPdRF.
I agree with Lucas' comment: showing every single puzzle and model is too specific and will be nearly impossible to keep up-to-date. The number of puzzle released every month is huge and we would have to review these puzzles before we can put them on the website. On the one hand, this takes a lot of time. On the other hand, my experiences with cubing manufacturers and their willingness to cooperate with the WCA are not that good.

Apart from that, the draft is obviously for the puzzle section only, so as Lucas already suggested, we should move that to #168.

**Comment by Laura-O** [comment]
Disclaimer 1: My plan was to write a quick draft for this issue but that has become a bit more comprehensive and might look like an application for the WST... I'm sorry about that, I just hope the screenshots will make everything clearer. :-)

Disclaimer 2: This is not meant as a todo-list for the WST but as a basis for discussion.

# WRC content on the WCA Website

## Overview: what we want to have (explanations below)
- Additional page linked in the Regulations menu in the navigation bar, with:
	- Short intro
	- Links to Regulations and Guidelines and short explanation
	- Link to Regulations translations and short explanation
	- Link to FAQ
	- Link to Hardware section (tbd)
	- Link to organizer guidelines
	- News section
- Embed WRC FAQs in [Website FAQ](https://www.worldcubeassociation.org/faq)
- Organizer guidelines
- Hardware page

## Additional page
To add all the content, we need a new page where we can structure and present everything. Naturally, this should be in the "Regulations" section of the navigation bar. For now, I have called this page "info".

As described above, the page has a short intro, links to the main regulation resources and then some WRC news.

![info](https://cloud.githubusercontent.com/assets/362249/25065328/7a7d1138-220e-11e7-94f2-7be2312775d3.png)

## FAQ
We have collected the most common questions we received via email and compiled them as FAQs. They are still WIP, but should be done soon.

There is already a FAQ section on the website. Adding another FAQ section for WRC questions would be bad: in the worst case, people who have general questions only read the WRC FAQs and vice-versa. Furthermore, some of the questions overlap or do not 100% fit into one category.

*My suggestion for a solution:* We can add the WRC questions to the general section. As we will probably end up with a total of 20-30 questions (including the existing ones), it would be nice to have an option to sort them in categories.

![faq](https://cloud.githubusercontent.com/assets/362249/25065338/9be9228a-220e-11e7-990e-e8e830d7660b.png)


## Hardware section
On the long run, we would also like to have a hardware section, which contains information about puzzles. This section should not focus on specific models, but rather on examples and edge cases (see [GitHub issue #168](https://github.com/thewca/wca-regulations/issues/168)).

The implementation of the hardware section will take more time, for content as well as for the design. So this section will probably be added later.

## Organizer Guidelines
The Organizer Guidelines are plain text, structured in several paragraphs. The page could be structured like the existing FAQ section. But as this is a continuous text and it should be read as a whole, it might be better to leave it as a full text (see screenshot for an example).

![organizer_guidelines](https://cloud.githubusercontent.com/assets/362249/25065341/a789b244-220e-11e7-8c9e-0e71707877ce.png)

## News
The news section should be similar to the announcements on the front page. I think we do not need a form/interface to post them as we will only add something every 2-3 months. However, it might be nice to have an option to archive them, so that old news are not shown on the page but are still available via a link.

![news](https://cloud.githubusercontent.com/assets/362249/25065375/3f5bd458-220f-11e7-8910-7bdf361dae18.png)

## Other issues: icons
- The FA icon for the Guidelines should be changed from 'question-circle' to 'sticky-note'. I never thought about the icon used for the Guidelines, but a question mark is not the best representation. Therefore, I suggest to changing this to 'sticky-note'.
- For the new info page, 'info' looks nice.
- For the organizer guidelines, 'check-square-o' could fit (though it's not linked in the navigation bar).

![navbar](https://cloud.githubusercontent.com/assets/362249/25065357/d493133e-220e-11e7-8795-d60111bf4dfb.png)

## Translations
- Organizer guidelines: they should be preferably translated, as they are particularly interesting for competitors from regions with no or few WCA competitions (mainly non-English). However, it's a lot of content, and I am not sure if the website translators are willing to do this.
- WRC FAQs: like the general FAQs, they should be translated. They are not too long and are not likely to be changed often.
- News: this is dynamic content and will (hopefully) be updated frequently, so I think we do not need a translation.

**Comment by viroulep** [comment]
I really like the overall structure and suggestions!

It looks like you directly existing controllers of the website, maybe the WST could simply take your work as a base, maybe split up chunks that are already good to be shipped, and polish/improve the rest?

**Comment by AlbertoPdRF** [comment]
I really like the structure of the section too! It seems like a very good starting point :)

**Comment by Laura-O** [comment]
> It looks like you directly existing controllers of the website, maybe the WST could simply take your work as a base, maybe split up chunks that are already good to be shipped, and polish/improve the rest?

Sure, it's already [here](https://github.com/Laura-O/worldcubeassociation.org/tree/WRC-section).

**Comment by lgarron** [comment]
I will probably want to clean up a few things when they land, but the overall approach looks pretty good!
